### ISSUE TITLE
Stop button not working

### ISSUE THREAD

**Issue by Sumientra** [SCIInstitute/SCIRun#2124]
**Describe the bug**
The stop buttons on modules and the main button for the whole network have no effect.

**In all bug cases, some combination of screenshots, sample network(s), and input data should be attached. You can use any cloud storage links you want for networks and data.**
[Here's](https://drive.google.com/open?id=1GKWH-IJ0VjnDIILU4M4YvzYNEuDZ39KH) a video of what happens. This is the simplest example I could put together, but it happens for me in all networks on all modules that take a long time to run.

**To Reproduce**
1. Run a network
2. Click on stop button
3. See error

**Expected behavior**
I expect the module, or the network, to stop executing.

**Version**
v5.0-beta.X-365-g57d4c2c19
(Video was made with an older version, but the issue still happens with the current version).

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version: 10.14.6

**Comment by dcwhite** [comment]
This has been broken for a while unfortunately. I'll investigate next week.

**Comment by dcwhite** [comment]
Re: the video--the network toolbar stop button is grayed out, so it will not do anything. I should probably just remove that one. ShowField did get the "Stoppable" feature, and that's what's broken; I'll try to fix that one first, but it's a module-by-module feature. Which other modules would you like me to attempt to fix?

**Comment by dcwhite** [comment]
It's a problem with the underlying thread library: https://stackoverflow.com/questions/30901904/os-x-libstdc-prevents-boostthread-from-interruptions. I will explore different recompilation options, but really I should reimplement stuff to use std::thread and a better stoppable idiom, like discussed here: https://thispointer.com/c11-how-to-stop-or-terminate-a-thread/. The problem with the current boost::thread code is if you do stop a ShowField, all the memory used by the module will not be cleaned up, and that will probably cause more problems than just killing the whole program and starting over.

**Comment by Sumientra** [comment]
Since you found an issue with the underlying library, should I still list specific modules to fix?

**Comment by dcwhite** [comment]
Yeah, I think I can code up a replacement in about a week.

**Comment by Sumientra** [comment]
I'm just running into a network getting stuck on a module and was wondering if there is any news on the stop buttons. I see that I was supposed to send a list of modules. Currently I am waiting on BuildFEMatrix. ShowField as mentioned is definitely useful. GetFieldBoundary and SolveLinearSystem would be nice too.

**Comment by dcwhite** [comment]
There's the first part of the implementation done in #2195 which was just merged in. The stop button "works" in the sense that it can stop processing now, but there's no error checking on it so it's quite unstable. I'll be working on polishing it up pretty soon.

**Comment by Sumientra** [comment]
Great. Looking forward to trying this out :)

**Comment by dcwhite** [comment]
The new implementation needs some rethinking, since it caused some bad slowdowns on Windows and Linux. So this feature is on hold for a bit.

**Comment by dcwhite** [comment]
**Analysis (2026-06-13)**

Stop button: first implementation landed (#2195) but caused bad slowdowns on Windows/Linux and was put on hold. Root cause was boost::thread interruption not working on macOS (libstdc++ issue). The comment suggests a std::thread-based reimplementation with a proper stoppable idiom. The ShowField 'stoppable' feature was the priority use case. Still unresolved.

**Comment by dcwhite** [comment]
Grouped under #2625.

Following up on the 2021-04 "needs some rethinking, caused bad slowdowns on Windows and Linux" note — the cause is identifiable. `Stoppable::stopRequested()` performs a `std::future::wait_for()` on every poll (`src/Core/Thread/Mutex.cc:94`), which is a mutex acquire and potentially a futex syscall rather than a flag read. That matches the platform ranking, since libstdc++ is worst. Commit `a90013193` then removed all interruption checks and the `checkForInterruption()` helper in response.

Filed as #2621: replace the promise/future with `std::atomic<bool>`, making the poll a relaxed load. Then #2622 restores the checks.

Also worth noting for whoever picks this up — `ModuleWidget::stopButtonPushed()` is currently `#if 0`'d out (`src/Interface/Application/ModuleWidget.cc:1472`, `//TODO: doesn't quite work yet`), so nothing can request a module stop from the GUI at present. Re-enabling that is the GUI half of this issue.

`SolveLinearSystem`, requested above, is covered by #2623 (live mid-solve steering) plus the `SolveLinearSystem` box in #987 (abort).
